### PR TITLE
Check [extras] when determining version compatibility

### LINKED ISSUE #110: Test-only dependencies in [compat] cause an error
See https://github.com/JuliaStats/Distributions.jl/issues/868#issuecomment-485991894. Distributions has a test dependency on JSON, and lists it in `[compat]` and `[extras]`. The fact that it isn't in `[deps]` makes the registration break.

### PR DESCRIPTION
**Pull request by ararslan**
Fixes #110.

### CODE REVIEW

**Comment by ararslan** [comment]
I have no idea whatsoever how to test this or how most of Registrator works, so some help and guidance would be appreciated.

**Comment by iamed2** [comment]
You'll want to add a check like https://github.com/JuliaComputing/Registrator.jl/blob/45d066138b2e9a3939a8c01f2aaef47371c2c584/src/regedit/register.jl#L262 for extras (just checks to see that they're available in the registries). 

Then you'll also want to run the `findpackageerror` part of the `else` clause right below what you've added.

IMO all the targets should be checked for version compatibility but this is fine for now.

**Comment by ararslan** [comment]
Is this more like what you have in mind, @iamed2?

File: src/regedit/register.jl [hunk 1]
```diff
@@ -348,18 +348,29 @@ function register(
...
-                uuidofdep = string(pkg.deps[n])
 
                 err = findpackageerror(n, uuidofdep, regdata)
                 if err !== nothing
                     @debug(err)
                     return RegBranch(pkg, branch; er=err)
                 end
+
+                if inextras && !indeps
+                    @debug("$n is a test-only dependency; omitting from Versions.toml")
```
Comment (iamed2) [inline src/regedit/register.jl]:
```suggestion
                    @debug("$n is a test-only dependency; omitting from Compat.toml")
```

**Comment by iamed2** [comment]
This is good, just one suggestion because this section reads Versions.toml but writes Compat.toml.

**Pull request merged**